pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -297,18 +297,11 @@ async def _publish(self, filename):
                     await asyncio.sleep(5)
 
     async def handle_failure(self):
-        # Increment failure count
-        current_count = await redis.get_value(self.fail_count_name)
-        if current_count is None:  # does not yet exist in Redis
-            current_count = 0
-        fail_count = int(current_count) + 1
-        self.runtime.logger.info('Failure count for %s: %s', self.version, fail_count)
-
         # Update fail counter on Redis
-        await redis.set_value(self.fail_count_name, fail_count)
+        fail_count = await redis.increment_counter(self.fail_count_name)
 
-        # Less than 2 failures, assembly != stream: just break the build
-        if fail_count < 2 or self.assembly != 'stream':
```

Comment:
Why did we change this?

Reply:
Line 393 already has
```
        if assembly == 'stream':
            await pipeline.handle_failure()
```
so no need to check that again